docs/trouble_shooting.rst

```diff
@@ -0,0 +1,109 @@
+
+   [user@exflonc12 ~]$ extra-foam DSSC SCS
+
+   services.py - WARNING - Found Redis server for DSSC (started at 2020-02-06 12:50:03.906872)
+   already running on this machine using port 6380!
+
+   You can choose to shut down the Redis server. Please note that the owner of the Redis server
+   will be informed (your username and IP address).
+
+   Shut down the existing Redis server? (y/n)
+
+**EXtra-foam** uses `Redis` as broker to pass meta information between different processes. By
+design, each type of detector has its unique `Redis` port so one can safely run more than one
+**EXtra-foam** instances for different detectors on the same machine. However, it is not allowed
+to run two instances with the same type of detector. Also, **EXtra-foam** receives data from
+**karabo bridge** and thus there can be data loss if there is any instance secretly running
+in the background, stealing the data.
+
+In the instrument control room, there should be only one **EXtra-foam** instance for the detector
+that is running. Therefore, it is safe to type "y" to shut down the existing *Redis* server.
+However, if somebody wants to make a joke about you and did that remotely, you will get informed.
+
+Config file is invalid
+++++++++++++++++++++++
+
+If you are prompted to warning like,
+
+.. code-block:: console
+
+   Traceback (most recent call last):
+     File "/home/username/anaconda3/envs/foam/bin/extra-foam", line 11, in <module>
+       load_entry_point('EXtra-foam', 'console_scripts', 'extra-foam')()
+     File "/home/username/xfel-data-analyais/EXtra-foam/extra_foam/services.py", line 356, in application
+       config.load(detector, topic)
+     File "/home/username/xfel-data-analyais/EXtra-foam/extra_foam/config.py", line 456, in load
+       self._data.load(detector, topic)
+     File "/home/username/xfel-data-analyais/EXtra-foam/extra_foam/config.py", line 382, in load
+       self.from_file(det, topic)
+     File "/home/username/xfel-data-analyais/EXtra-foam/extra_foam/config.py", line 393, in from_file
+       raise OSError(msg)
+   OSError: Invalid config file: /home/username/.EXtra-foam/scs.config.yaml
+   ParserError('while parsing a block mapping', <yaml.error.Mark object at 0x7fcffbd84910>,
+   "expected <block end>, but found '<block mapping start>'", <yaml.error.Mark object at 0x7fcffbd84ed0>)
+
+This error is triggered when the :ref:`config file` is not valid. Please correct it if you have modified
+the default one. Alternatively, you can delete it and let the program generate a default one for you.
+
+No data is received
++++++++++++++++++++
+
+If **EXtra-foam** opens up fine and running it by clicking on *start* button does
+nothing, please make sure that relevant **PipeToZeroMQ** device is properly
+configured, activated and its *data sent* property is updating. This device
+can be configured only with the help of experts (data analysis support and beamline scientists).
+
+Incorrect dependencies
+++++++++++++++++++++++
+
```

Comment:
@RobertRosca You are welcome to comment or add something for this part. Thank you!